lib/jelix/core/jRequest.class.php

```diff
@@ -386,48 +386,51 @@ function getPort($forceHttps = null) {
    }
 
    /**
-    * call it when you want to read the content of the body of a request
-    * when the method is not GET or POST
-    * @return mixed    array of parameters or a single string when the content-type is unknown
-    * @since 1.2
-    */
-   public function readHttpBody() {
-      $input = file_get_contents("php://input");
-      $values = array();
-
-      if (!isset($_SERVER["CONTENT_TYPE"])) {
-        return $input;
-      }
-      if (strpos($_SERVER["CONTENT_TYPE"], "application/x-www-form-urlencoded") === 0) {
-         parse_str($input, $values);
-         return $values;
-      }
-      else if (strpos($_SERVER["CONTENT_TYPE"], "multipart/form-data") === 0) {
+     * call it when you want to read the content of the body of a request
+     * when the method is not GET or POST
+     * @return mixed    array of parameters or a single string when the content-type is unknown
+     * @since 1.2
+     */
+    public function readHttpBody() {
+        $input = file_get_contents('php://input');
 
-         if (!preg_match("/boundary=([a-zA-Z0-9]+)/", $_SERVER["CONTENT_TYPE"], $m))
+        if (!isset($_SERVER['CONTENT_TYPE'])) {
             return $input;
+        }
+
+        $values = array();
+
+        if ($_SERVER['CONTENT_TYPE'] == 'application/x-www-form-urlencoded') {
```

Comment:
I prefer to test with strpos: the content-type field may content also the (optional) charset. I will change that